azurelinuxagent/ga/exthandlers.py

```diff
@@ -486,33 +491,44 @@ def handle_enable(self, ext_handler_i):
             ext_handler_i.update_settings()
 
         ext_handler_i.enable()
+        # Remove the UNINSTALL_FAILED from env variables list after calling Install and Enable
+        ext_handler_i.remove_key_from_handler_env(UNINSTALL_FAILED)
 
     @staticmethod
     def _update_extension_handler(old_ext_handler_i, ext_handler_i):
 
-        def execute_old_handler_command(func):
+        def execute_old_handler_command(op, func):
             """
             Created a common wrapper to execute all commands that need to be executed from the old handler
             so that it can have a common exception handling mechanism
+            :param op: The current running operation
             :param func: The command to be executed on the old handler
             """
+            continue_on_update_failure = False
             try:
+                continue_on_update_failure = ext_handler_i.load_manifest().is_continue_on_update_failure()
```

Comment:
I just realized that, on each extension command, we call `load_manifest` which in turn always reads the file from disk and parses it. I don't see why we would have to parse the same file over and over again. Can we have a method that will, when called the first time, parse the file, and when called subsequently just return the parsed contents?

Reply:
Good point while doing this code I felt the same way. We can definitely make it part of the ExtHandlerInstance or ExtHandler object to avoid the extra I/O. 
That would be another PR though as this issue is outside the scope of this problem :)